target_chains/solana/programs/pyth-solana-receiver/tests/common/mod.rs

```diff
@@ -169,37 +87,61 @@ pub fn build_merkle_root_encoded_vaa(
     }
 }
 
+pub fn build_guardian_set_account() -> Account {
+    let guardian_set = GuardianSet {
+        index:           DEFAULT_GUARDIAN_SET_INDEX,
+        keys:            create_dummy_guardians()
+            .iter()
+            .map(|x| {
+                let mut result: [u8; 20] = [0u8; 20];
+                result.copy_from_slice(
+                    &keccak::hashv(&[&PublicKey::from_secret_key(x).serialize()[1..]]).0[12..],
+                );
+                result
+            })
+            .collect::<Vec<[u8; 20]>>(),
+        creation_time:   0.into(),
+        expiration_time: 0.into(),
+    };
 
-pub struct ProgramTestFixtures {
-    pub program_simulator:    ProgramSimulator,
-    pub encoded_vaa_address:  Pubkey,
-    pub merkle_price_updates: Vec<MerklePriceUpdate>,
+    let guardian_set_data = (
+        <GuardianSet as anchor_lang::Discriminator>::DISCRIMINATOR,
+        guardian_set,
+    )
+        .try_to_vec()
+        .unwrap();
+
+    Account {
+        lamports:   Rent::default().minimum_balance(guardian_set_data.len()),
+        data:       guardian_set_data,
+        owner:      BRIDGE_ID,
+        executable: false,
+        rent_epoch: 0,
+    }
 }
 /**
  * Setup to test the Pyth Receiver. The return values are a tuple composed of :
  * - The program simulator, which is used to send transactions
- * - The pubkey of an encoded VAA account, which is pre-populated and can be used to test post_updates
- * - A vector of MerklePriceUpdate, corresponding to that VAA
+ * - The pubkeys of the encoded VAA accounts corresponding to the VAAs passed as argument, these accounts are prepopulated and can be used to test post_updates
  */
-pub async fn setup_pyth_receiver(
-    price_feed_messages: Vec<PriceFeedMessage>,
-) -> ProgramTestFixtures {
+pub async fn setup_pyth_receiver(vaas: Vec<Vec<u8>>) -> ProgramTestFixtures {
```

Comment:
I suggest passing the VAA accounts as `RawMessage` (or whatever the right wormhole type is) instead of `Vec<u8>` for clarity. Someone coming to this code is going to have a hard time figuring out how to populate these vectors of bytes.

Reply:
Done